files/num_occurrences_filter.rb

```diff
@@ -0,0 +1,106 @@
+#
+# This code will run inside sensu-server process. Be careful with external calls
+# that could disrupt event loop.
+#
+module Sensu::Extension
+  class NumOccurrences < Filter
+
+    STOP_PROCESSING  = 0
+    ALLOW_PROCESSING = 1
+
+    def name
+      'num_occurrences_filter'
+    end
+
+    def description
+      'filter events based on the number of occurrences'
+    end
+
+    def run(event)
+      begin
+        rc, msg = filter_by_num_occurrences(event)
+        yield msg, rc
+      rescue => e
+        # filter crashed - let's pass this on to handler
```

Comment:
I can't find an easy way to get a reference to Sensu::Server::Process object from here, which makes logging hard.

Also, unfortunately, upstream doesn't log message - it just eats output at https://github.com/sensu/sensu/blob/master/lib/sensu/server/filter.rb#L240